### PR TITLE
fix: remove email requirement for the user, and use the username if no email provided

### PR DESCRIPTION
**Pull request by gauthier-th**
#### Description

This PR remove the email requirement for users, but only from the user's perspective. From server side, Jellyfin/Plex username will be used when no email is provided, since a lot of Jellyseerr features need this email field.

#### To-Dos

- [x] Successful build `pnpm build`
- [x] Translation keys `pnpm i18n:extract`
- [ ] Database migration (if required)

### CODE REVIEW

File: src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx [hunk 1]
```diff
@@ -93,9 +93,14 @@ const UserGeneralSettings = () => {
...
 
   const UserGeneralSettingsSchema = Yup.object().shape({
-    email: Yup.string()
-      .email(intl.formatMessage(messages.validationemailformat))
-      .required(intl.formatMessage(messages.validationemailrequired)),
+    email:
+      user?.id === 1
+        ? Yup.string()
+            .email(intl.formatMessage(messages.validationemailformat))
+            .required(intl.formatMessage(messages.validationemailrequired))
```
Comment (Jacouille) [inline src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx:100]: I'd recommend swapping these two as we'd like to show the required message when the field is empty. Currently, the email validation will be displayed when the field is empty

File: server/routes/auth.ts [hunk 2]
```diff
@@ -320,15 +320,18 @@ authRoutes.post('/jellyfin', async (req, res, next) => {
...
+        email: body.email || account.User.Name,
         jellyfinUsername: account.User.Name,
         jellyfinUserId: account.User.Id,
         jellyfinDeviceId: deviceId,
         jellyfinAuthToken: account.AccessToken,
         permissions: Permission.ADMIN,
         avatar: account.User.PrimaryImageTag
           ? `${jellyfinHost}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`
-          : gravatarUrl(body.email ?? '', { default: 'mm', size: 200 }),
+          : gravatarUrl(body.email || account.User.Name, {
```
Comment (fallenbagel) [inline server/routes/auth.ts:331]: Why was this added? Afaik gravatar only uses emails

File: server/routes/auth.ts [hunk 3]
```diff
@@ -371,7 +374,7 @@ authRoutes.post('/jellyfin', async (req, res, next) => {
       if (account.User.PrimaryImageTag) {
         user.avatar = `${jellyfinHost}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`;
       } else {
-        user.avatar = gravatarUrl(user.email, {
+        user.avatar = gravatarUrl(user.email || account.User.Name, {
```
Comment (fallenbagel) [inline server/routes/auth.ts:377]: See above

File: server/routes/auth.ts [hunk 4]
```diff
@@ -426,7 +425,10 @@ authRoutes.post('/jellyfin', async (req, res, next) => {
         permissions: settings.main.defaultPermissions,
         avatar: account.User.PrimaryImageTag
           ? `${jellyfinHost}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`
-          : gravatarUrl(body.email, { default: 'mm', size: 200 }),
+          : gravatarUrl(body.email || account.User.Name, {
```
Comment (fallenbagel) [inline server/routes/auth.ts:428]: See above

File: server/routes/user/index.ts [hunk 5]
```diff
@@ -108,7 +121,7 @@ router.post(
       }
 
       const passedExplicitPassword = body.password && body.password.length > 0;
-      const avatar = gravatarUrl(body.email, { default: 'mm', size: 200 });
+      const avatar = gravatarUrl(email, { default: 'mm', size: 200 });
```
Comment (fallenbagel) [inline server/routes/user/index.ts:124]: Here too

**Review by Fallenbagel — CHANGES_REQUESTED** [review]

File: server/routes/auth.ts (hunk 2 above)
Comment (gauthier-th) [reply server/routes/auth.ts:331]: Yep, but the `gravatar-url` package throws an error if the provided string is empty, but send a default profile picture when the provided a string/email doesn't have a corresponding profile picture.
I could replace the `|| account.User.Name` by something like `|| 'none'` if you thing it makes more sense.

Comment (fallenbagel) [reply server/routes/auth.ts:331]: Email will always be there though. And if it's not there an empty string would give the default image (atleast it used to. Does it not now?)

Comment (gauthier-th) [reply server/routes/auth.ts:331]: Email should be here yes. I added this because during one of my test I came across an empty email and `gravatar-url` throwed an error because of it.
See https://github.com/sindresorhus/gravatar-url/blob/main/index.js#L4-L6

Comment (fallenbagel) [reply server/routes/auth.ts:331]: Ah makes sense. I assumed since overseerr was also just passing in email only

**Review by Fallenbagel — APPROVED** [review]
